Add data sharing info (#28)
neuroinformatics-unit/HowTo: docs/source/guides/Data-sharing.md

# Data Sharing

## Background

> Data is key to neuroscience, and it should be shared for maximal utility. There is increasing pressure from funders and publishers to do so alongside publications:

Reviewer: ```suggestion
Data is key to neuroscience, and it should be shared for maximal utility. There is increasing pressure from funders and publishers to share data alongside publications:
```

**Wellcome:**

> _“We expect our researchers to maximise the availability of research data, software and materials with as few restrictions as possible. As a minimum, the data underpinning research papers should be made available to other researchers at the time of publication, as well as any original software that is required to view datasets or to replicate analyses.”_

Reviewer 2: Minor stylistic-pedantic issue. For this and following quotes, we could use the "official" markdown quote syntax with ">", for example:
```md
**Niko:**
> nothing useful
```
I think it will render nicely with Sphinx

Author: Thanks!

**Cell Press:**

_“All datasets that are composed of standardized datatypes and reported in Cell Press papers must be deposited in a datatype-specific, Cell-Press-recommended repository before a paper is accepted for publication.”_

_“All original code must be either deposited in a repository that mints DOIs or included in the supplemental information before a paper is accepted for publication.”_

## Requirements

Currently, SWC does not have any specific code and data sharing requirements, but your funder or publisher may do. 

**Ideally, all data and code will be shared in such a way that any other researcher could easily reproduce key figures from a publication, and build on those data and software in their own research.**

## Code

> Original code can be hosted in a public GitHub/GitLab repository, hosted on an organisational account (i.e. not an individual account). This could be the [SWC organisation](https://github.com/SainsburyWellcomeCentre), or that of a particular team, lab or collaboration. Code should include an [appropriate license](https://howto.neuroinformatics.dev/guides/Licensing.html), and may include a [DOI](https://www.doi.org/) (e.g. using [Zenodo](https://docs.github.com/en/repositories/archiving-a-github-repository/referencing-and-citing-content)). 

Reviewer 2: Personally, I would use stronger language than "may" for recommending attaching persistent identifiers to code repositories on GitHub/GitLab, especially when the code is associated with a manuscript. Given that the repo owner can at any time change or delete the code, without versioned archival on Zenodo or elsewhere, the version of the code associated with a paper is likely to be lost.
That said, we might be overreaching with it. Not everyone needs to hear my strong opinions on this.

Reviewer 2: I'd say it's worth making the distinction between making code public (which is what GitHub does) vs archiving for posterity (what Zenodo does). Doing the first doesn't automatically accomplish the second.

Author: I've strengthened it slightly with:
> To ensure a permanent archive of the code associated with a publication a [DOI](https://www.doi.org/) should be created (e.g. using [Zenodo](https://docs.github.com/en/repositories/archiving-a-github-repository/referencing-and-citing-content)).

I'm cautious of recommending too many things. If researchers did all the good/code data practices we would like, there may not be any time left to generate data!

## Data
### Repository
There are [many repositories](repo-table) that will accept the types of data generated by SWC. They come into three categories, general, neuroscience-specific and modality-specific.  If you do not have a preference, we recommended:

**For small (<20GB) datasets:**
Use the existing [SWC section of the UCL FigShare instance](https://rdr.ucl.ac.uk/The_Sainsbury_Wellcome_Centre). This is easy to use, and is supported centrally by UCL

**For larger datasets:**

> Create a GIN repository, ideally within the [SWC organisation](https://gin.g-node.org/SainsburyWellcomeCentre). GIN uses Git and provides a versioning system, making it particularly well suited for ongoing research projects.

Reviewer 2: Missing link to GIN?

Author: There's a link to the SWC org?

Reviewer 2: Ah yeah, sure.

**Specific modalities:**
In some cases, there may be a modality or field-specific repository that would be best suited for the data (or mandated by the publisher), some examples include:
- [Brain Image Library](http://www.brainimagelibrary.org/) - Large brain image datasets

> - [DANDI](http://www.brainimagelibrary.org/) - Electrophysiology, optophysiology, behavioural time-series (must be in NWB format)

Reviewer 3: Wrong link for Dandi and BioImage Archive

### Structure
There is no single best way to structure your data, but it must be in a form that allows users to easily re-use it for their own purposes. This may be by adopting specific file formats (such as [NeuroData Without Borders](https://www.nwb.org/)). Alternatively the data could be organised within a standardised project folder structure. Many labs have their own system, but those starting from scratch may wish to use the [SWC-Blueprint data structure](https://swc-blueprint.neuroinformatics.dev/).

**However the data is structured, all relevant metadata must be included with the raw data.**

## Assistance
If you have any technical questions about sharing data or code, please contact adam.tyson@ucl.ac.uk.

(repo-table)=
## Reference of potential repositories for neuroscience data
| Repository                         |URL                                                |Domain                                                                                               |
|------------------------------------|---------------------------------------------------|-----------------------------------------------------------------------------------------------------|

> | Globus                             |                                                   |Anything                                                                                             |

Reviewer 2: Missing link for globus?

Author: There isn't a link for Globus really, it's a way of sharing data we have internally. Useful for collaborations, but not really for public sharing. It's here for completeness really.
